modules/core/src/main/java/org/apache/ignite/internal/cluster/DistributedTransactionConfiguration.java

```diff
@@ -0,0 +1,334 @@
+
+    /**
+     * @param ctx Kernal context.
+     * @param log Log.
+     * @param longOperationsDumpTimeoutLsnr Listener of {@link #longOperationsDumpTimeout} change event.
+     * @param collisionsDumpIntervalLsnr Listener of {@link #collisionsDumpInterval} change event.
+     */
+    public DistributedTransactionConfiguration(
+        GridKernalContext ctx,
+        IgniteLogger log,
+        DistributePropertyListener<Long> longOperationsDumpTimeoutLsnr,
+        DistributePropertyListener<Integer> collisionsDumpIntervalLsnr
+    ) {
+        ctx.internalSubscriptionProcessor().registerDistributedConfigurationListener(
+            new DistributedConfigurationLifecycleListener() {
+                @Override public void onReadyToRegister(DistributedPropertyDispatcher dispatcher) {
+                    txOwnerDumpRequestsAllowed.addListener(makeUpdateListener(PROPERTY_UPDATE_MESSAGE, log));
+                    longOperationsDumpTimeout.addListener(makeUpdateListener(PROPERTY_UPDATE_MESSAGE, log));
+                    longTransactionTimeDumpThreshold.addListener(makeUpdateListener(PROPERTY_UPDATE_MESSAGE, log));
+                    transactionTimeDumpSamplesCoefficient.addListener(makeUpdateListener(PROPERTY_UPDATE_MESSAGE, log));
+                    longTransactionTimeDumpSamplesPerSecondLimit.addListener(makeUpdateListener(PROPERTY_UPDATE_MESSAGE, log));
+                    collisionsDumpInterval.addListener(makeUpdateListener(PROPERTY_UPDATE_MESSAGE, log));
+                    longOperationsDumpTimeout.addListener(longOperationsDumpTimeoutLsnr);
+                    collisionsDumpInterval.addListener(collisionsDumpIntervalLsnr);
+
+                    dispatcher.registerProperties(txOwnerDumpRequestsAllowed, longOperationsDumpTimeout,
+                            longTransactionTimeDumpThreshold, transactionTimeDumpSamplesCoefficient,
+                            longTransactionTimeDumpSamplesPerSecondLimit, collisionsDumpInterval);
+                }
+
+                @Override public void onReadyToWrite() {
+                    if (allNodesSupport(ctx, DISTRIBUTED_METASTORAGE)) {
+                        setDefaultValue(longOperationsDumpTimeout, dfltLongOpsDumpTimeout, log);
+                        setDefaultValue(longTransactionTimeDumpThreshold, dfltLongTransactionTimeDumpThreshold, log);
+                        setDefaultValue(transactionTimeDumpSamplesCoefficient, dfltTransactionTimeDumpSamplesCoefficient, log);
+                        setDefaultValue(longTransactionTimeDumpSamplesPerSecondLimit, dfltLongTransactionTimeDumpSamplesPerSecondLimit, log);
+                        setDefaultValue(collisionsDumpInterval, dfltCollisionsDumpInterval, log);
+                        setDefaultValue(txOwnerDumpRequestsAllowed, dfltTxOwnerDumpRequestsAllowed, log);
+                    } else {
+                        longOperationsDumpTimeout.localUpdate(dfltLongOpsDumpTimeout);
```

Comment:
let's add some warning message here, like `log.warning("Distributed metastorage is not supported." + some useful info about local update`